resolwe/flow/managers/dispatcher.py

```diff
@@ -531,6 +538,32 @@ def _prepare_context(self, data_id, data_dir, runtime_dir, **kwargs):
             if k.startswith("STATUS_") and isinstance(getattr(Data, k), str)
         }
 
+        # Missing data settings.
+        missing_data = self._get_missing_data(data)
```

Comment:
Some data may be removed between this step and when the executor is started. It would make sense to add a lock here. But then we would need a way to remove stale locks - e.g., when scheduler crashes

Reply:
I believe it would be best to request this information from manager when executor is starting. And maybe touch all available data locations just to make sure they do not get deleted while missing data is being transfered.

Reply:
I agree it would be best to require locks on existing data and get the list of missing data at the same time.